src/uriHandler.ts

```diff
@@ -25,17 +26,23 @@ export class UriEventHandler extends vscode.EventEmitter<vscode.Uri> implements
     return UriEventHandler.instance;
   }
 
-  public handleUri(uri: vscode.Uri) {
-    switch (uri.path) {
+  public async handleUri(uri: vscode.Uri) {
+    const { authority, path, query } = uri;
+    switch (path) {
       case "/authCallback":
         logger.debug("Got authCallback URI, firing as Event", uri);
         this.fire(uri);
         break;
       case "/consume":
         vscode.commands.executeCommand("confluent.topic.consume.fromUri", uri);
         break;
-      default:
-        logger.warn("Got unexpected URI, ignoring", uri);
+      case "/projectScaffold":
+        if (authority === "confluentinc.vscode-confluent") {
+          projectScaffoldUri.fire(uri); // Emit the event
+        } else {
+          vscode.window.showErrorMessage("Unsupported URI path.");
+        }
```

Comment:
I think we can safely remove this `authority` check since VS Code shouldn't route any other URIs through our extension. Were you able to get that error notification to show up?

Reply:
no--- which is good! & good point, will remove